project/npda/tests/form_tests/test_patient_form.py

```diff
@@ -0,0 +1,298 @@
+# Standard imports
+from enum import Enum
+import pytest
+import logging
+from unittest.mock import Mock, patch
+
+# 3rd Party imports
+from django.core.exceptions import ValidationError
+from dateutil.relativedelta import relativedelta
+from requests import RequestException
+
+# NPDA Imports
+from project.npda.models.patient import Patient
+from project.npda.forms.patient_form import PatientForm
+from project.npda import general_functions
+from project.npda.tests.factories.patient_factory import (
+    TODAY,
+    DATE_OF_BIRTH,
+    VALID_FIELDS,
+    VALID_FIELDS_WITH_GP_POSTCODE,
+    INDEX_OF_MULTIPLE_DEPRIVATION_QUINTILE
+)
+
+# Logging
+logger = logging.getLogger(__name__)
+
+
+# We don't want to call remote services in unit tests
+@pytest.fixture(autouse=True)
+def mock_remote_calls():
+    with patch("project.npda.forms.patient_form.validate_postcode", Mock(return_value=True)):
```

Comment:
I tried reducing the nesting here with an [ExitStack](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) but it didn't work. There's probably a way to do it but it's not that bad and wrapped up in a fixture anyway